### PR TITLE
[Fix] plfq cornercases and exp. design checks

### LINKED ISSUE #5224: ProteomicsLFQ exits with segfault *in the end* when experimental design numbering is not correct
I.e. not consecutive or not starting from 1.
We have seen cases where it worked but this was probably lucky.
This needs to be safer. And even if it cant be made much safer since we need to assume that ordering, then we need to check very carefully and very early, already during reading, or better the construction of an experimental design object.

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

addresses #5221 partly
fixes #5224

### CODE REVIEW

File: src/openms/source/METADATA/ExperimentalDesign.cpp [hunk 1]
```diff
@@ -599,8 +599,55 @@ namespace OpenMS
...
+      {
+        throw Exception::InvalidValue(
+            __FILE__,
+            __LINE__,
+            OPENMS_PRETTY_FUNCTION, "Fraction groups do not start with 1.", String(msfile_section_[0].fraction_group));
+      }
+
+      Size last_fraction = 1;
+      Size last_fraction_group = 0;
       for (const MSFileSectionEntry& row : msfile_section_)
```
Comment (timosachsenberg) [inline src/openms/source/METADATA/ExperimentalDesign.cpp:619]: add comment above for loop:
// ensure that fraction group and fraction identifier are consecutive numbers

File: src/openms/source/METADATA/ExperimentalDesign.cpp [hunk 2]
```diff
@@ -599,8 +599,55 @@ namespace OpenMS
       std::set< std::tuple< std::string, unsigned > > path_label_set;
       std::map< std::tuple< unsigned, unsigned >, std::set< unsigned > > fractiongroup_label_to_sample;
 
+      if (msfile_section_[0].fraction != 1)
```
Comment (timosachsenberg) [inline src/openms/source/METADATA/ExperimentalDesign.cpp:602]: add comment above:
// ensure that fraction group and fraction identifiers start with 1

File: src/openms/source/METADATA/ExperimentalDesign.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/source/METADATA/ExperimentalDesign.cpp:619]: isnt that obvious from the exception?

File: src/openms/source/METADATA/ExperimentalDesign.cpp (hunk 2 above)
Comment (jpfeuffer) [reply src/openms/source/METADATA/ExperimentalDesign.cpp:602]: see other

File: src/utils/ProteomicsLFQ.cpp [hunk 3]
```diff
@@ -1709,7 +1709,7 @@ class ProteomicsLFQ :
 
         if (e != EXECUTION_OK) { return e; }
         
-        if (getStringOption_("transfer_ids") != "false")
+        if (getStringOption_("transfer_ids") != "false" && ms_files.second.size() > 1)
         {
```
Comment (timosachsenberg) [inline src/utils/ProteomicsLFQ.cpp:1713]:
```suggestion
        {  // transfer IDs between multiple ms files
```

File: src/openms/source/METADATA/ExperimentalDesign.cpp [hunk 4]
```diff
@@ -599,8 +599,55 @@ namespace OpenMS
...
+      if (msfile_section_[0].fraction_group != 1)
+      {
+        throw Exception::InvalidValue(
+            __FILE__,
+            __LINE__,
+            OPENMS_PRETTY_FUNCTION, "Fraction groups do not start with 1.", String(msfile_section_[0].fraction_group));
+      }
+
+      Size last_fraction = 1;
+      Size last_fraction_group = 0;
```
Comment (timosachsenberg) [inline src/openms/source/METADATA/ExperimentalDesign.cpp:618]:
```suggestion
      Size last_fraction_group = 0;
      // ensure that fraction group and fraction identifier are consecutive numbers
```

File: src/openms/source/METADATA/ExperimentalDesign.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/source/METADATA/ExperimentalDesign.cpp:619]: and the loop does not only do that

File: src/openms/source/METADATA/ExperimentalDesign.cpp (hunk 2 above)
Comment (timosachsenberg) [inline src/openms/source/METADATA/ExperimentalDesign.cpp:602]:
```suggestion
      // ensure that fraction group and fraction identifiers start with 1
      if (msfile_section_[0].fraction != 1)
```

**Review by timosachsenberg — APPROVED** [review]

**Pull request merged**